.github/workflows/ci.yml

```diff
@@ -190,8 +190,15 @@ jobs:
           version: ${{ env.DOCKER_BUILDX_VERSION }}
           install: true
 
+      - name: Login to ghcr.io
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Since this is the default actions token I believe we need an extra permissions block for the workflow with write:packages 

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

Reply:
thanks for pointing that out @jastang, I'm looking into that more now. looks like I would need to add something like below to this `publish-artifacts` job:
```
  publish-artifacts:
    permissions:
      packages: write
```

Any idea why that permissions granting doesn't seem to be present in these new workflows that the upjet providers are using though? 🤔 

* https://github.com/crossplane-contrib/provider-upjet-aws/blob/main/.github/workflows/publish-provider-packages.yaml
* https://github.com/crossplane-contrib/provider-workflows/blob/main/.github/workflows/publish-provider-family.yml

Reply:
ah, maybe because https://github.com/crossplane-contrib/provider-upjet-aws/settings/actions has `Read and write permissions` set:
<img width="786" alt="Screenshot 2025-02-23 at 11 19 48 AM" src="https://github.com/user-attachments/assets/2a1be91d-00c9-4cf2-bfb0-4f977d2bcdcd" />

provider-nop has that set also, so this could work without the explicit granting of permissions in the workflow file.

Reply:
Ah maybe that's sufficient then 👍 I saw those were using a different token than the default (GH_PAT). I think I prefer this more explicit approach though.

Reply:
I think that `secrets.GHCR_PAT` is just passed in by the caller using regular ol' `${{ secrets.GITHUB_TOKEN }}`. So i think those workflows are using the default token too.

>  I think I prefer this more explicit approach though

are you saying you'd like to see the explicit declaration of `packages: write` in this workflow? happy to make that change if that's what you're saying 🤓

Reply:
I'm going to go ahead and merge and we can follow up with another PR if you'd like @jastang! (we may need to follow up anyways if this doesn't work as intended in `main` 🙏)

Reply:
Good point re: workflow caller, I missed that. And no, I was referring to "hiding" permissions in a separately minted token versus granting it to the default token - but I was mistaken about that. This all looks fine to me -- though it seems my approval isn't enough 😇

Reply:
nice, it looked like it worked in main!

https://github.com/crossplane-contrib/provider-nop/pkgs/container/provider-nop/361188723?tag=v0.4.0-rc.0.3.gec625eb

```
❯ docker pull ghcr.io/crossplane-contrib/provider-nop:v0.4.0-rc.0.3.gec625eb
v0.4.0-rc.0.3.gec625eb: Pulling from crossplane-contrib/provider-nop
94be777fcdbc: Already exists
2d61dea2ec5c: Pull complete
144f6724f85f: Pull complete
bbc66066145c: Pull complete
Digest: sha256:df8e9ab610cdc4764d3f38a560514aecdeb97db388aa89083c4cfb8e6da5784a
Status: Downloaded newer image for ghcr.io/crossplane-contrib/provider-nop:v0.4.0-rc.0.3.gec625eb
ghcr.io/crossplane-contrib/provider-nop:v0.4.0-rc.0.3.gec625eb
```